modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/input_move_boxes.js

```diff
@@ -283,11 +289,23 @@ AUI.add(
 						var btnDisabledRight = true;
 
 						if (target.get('length') > 0) {
-							if (target == instance._rightBox) {
-								btnDisabledLeft = false;
+							if (target === instance._rightBox) {
```

Comment:
Also, looking at this now a bit more, I don't think we need the branching, just a way to pick which `maxItems` setting to take and then the algorithm is all the same.

I'll push a final commit later today simplifying that, unless @markocikos beats me to it 😘

Reply:
Instead of defaulting max items to `-1`, why don't you default it to `Infinity`?

Then you can get rid of the whole `leftBoxMaxItems === -1 ||` and just leave the `selectedOptions <= leftBoxMaxItems`:

```js
const btnDisabledLeft = selectedOptions <= leftBoxMaxItems;

const btnDisabledRithg = selectedOptions <= rightBoxMaxItems;
```

As you said, BRANCHING IS FOR LOSERS.

![what-a-loser](https://user-images.githubusercontent.com/7074/87669922-196b9680-c76f-11ea-9a6d-9e8aba1ad8a0.gif)

Reply:
> Instead of defaulting max items to -1, why don't you default it to Infinity?

I was considering using `-1` instead of `Infinity` due to JSP default value for these properties